evm_loader/program/src/executor/state.rs

```diff
@@ -431,13 +433,15 @@ impl<'a, B: AccountStorage> Database for ExecutorState<'a, B> {
         Ok(cache.block_timestamp)
     }
 
+    #[allow(clippy::too_many_lines)]
     async fn external_account(&self, address: Pubkey) -> Result<OwnedAccountInfo> {
         self.touch_solana(address);
-
+        // collect accounts for actions that we don't processed yet
         let metas = self
             .data
             .actions
             .iter()
+            .skip(self.data.cache.borrow_mut().actions_offset)
```

Comment:
Please only use `self.data.cache.borrow_mut()` once per function.
`let mut cache = self.data.cache.borrow_mut();`

Clippy may display a warning about holding a reference over an 'await' point.
This is a false positive and can be suppressed.